demos/jobs/jobs_demo_mosquitto/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+set( DEMO_NAME "jobs_demo_mosquitto" )
+
+set_source_files_properties( "${DEMO_NAME}.c" PROPERTIES COMPILE_FLAGS "-Wno-unused-parameter" )
```

Comment:
What is the reasoning for disabling the `-Wunused-parameter` flag? 
If there is a valid reason, then a comment stating the reason would be good to have

Reply:
The callbacks for libmosquitto have parameters that are not used. Comment added.